### PR TITLE
fix: Invalid cron schedule does not include positional information

### LINKED ISSUE #1629: Invalid cron schedule does not include positional information
I typed in an erroneous cron schedule `0 0 0 0 0` and got:

> error: build and deploy failed for module "invoicing": failed to build module "invoicing": failed to extract module schema: value 0 out of allowed day of month range of 1-31

The LSP wasn't able to do anything with this, so as a user I didn't know there was an error until I looked at the FTL output, largely defeating the purpose of the VSCode extension.

Two tasks:

1. The schema for the cron task has positional information, so it should be surfaced.
2. In the more general case of errors with no positional information, the VSCode extension should pop up a notification.

### PR DESCRIPTION
**Pull request by gak**
#1629

Metadata positioning was not being populated.

image

image

image

### CODE REVIEW

File: go-runtime/compile/parser.go [hunk 1]
```diff
@@ -199,23 +242,27 @@ func parseDirectives(node ast.Node, fset *token.FileSet, docs *ast.CommentGroup)
 			continue
 		}
 		pos := fset.Position(line.Pos())
+		ppos := schema.Position{
+			Filename: pos.Filename,
+			Line:     pos.Line,
+			Column:   pos.Column + 2, // Skip "//"
+		}
+
 		// TODO: We need to adjust position information embedded in the schema.
```
Comment (matt2e) [inline go-runtime/compile/parser.go:251]: I think this old TODO can be removed now

File: internal/cron/cron.go [hunk 2]
```diff
@@ -55,11 +55,11 @@ func Next(pattern Pattern, allowCurrentTime bool) (time.Time, error) {
 	return NextAfter(pattern, time.Now().UTC(), allowCurrentTime)
 }
 
-// NextAfter calculcates the next time that matches the pattern after the origin time
+// NextAfter calculates the next time that matches the pattern after the origin time
 // If inclusive is true, the origin time is considered a valid match
 // All calculations are done in UTC, and the result is returned in UTC
 func NextAfter(pattern Pattern, origin time.Time, inclusive bool) (time.Time, error) {
-	// set original to the first acceptable time, irregardless of pattern
```
Comment (matt2e) [inline internal/cron/cron.go:62]: thank you 🤦‍♂️

**Review by matt2e — APPROVED** [review]

File: internal/cron/cron.go (hunk 2 above)
Comment (gak) [reply internal/cron/cron.go:62]: Ahh of course!

File: go-runtime/compile/parser.go (hunk 1 above)
Comment (gak) [reply go-runtime/compile/parser.go:251]: Ahh of course! :)

Comment (gak) [reply go-runtime/compile/parser.go]: `Adjust the Participle-reported` comment too!

File: backend/schema/validate.go [hunk 3]
```diff
@@ -540,7 +540,7 @@ func validateVerbMetadata(scopes Scopes, module *Module, n *Verb) (merr []error)
 		case *MetadataCronJob:
 			_, err := cron.Parse(md.Cron)
 			if err != nil {
-				merr = append(merr, err)
+				merr = append(merr, errorf(md, "verb %s: invalid cron expression %q: %v", n.Name, md.Cron, err))
```
Comment (alecthomas) [inline backend/schema/validate.go:543]: Aha!

**Review by alecthomas — APPROVED** [review]

**Pull request merged**